src/vue/pages/IncomingWithdrawalRequests.vue

```diff
@@ -0,0 +1,368 @@
+
+import { mapGetters } from 'vuex'
+import { vuexTypes } from '@/vuex'
+
+import { Bus } from '@/js/helpers/event-bus'
+import { ErrorHandler } from '@/js/helpers/error-handler'
+
+import WithdrawalRequestDetails from '@/vue/pages/withdrawals/WithdrawalRequestDetails'
+import { base } from '@tokend/js-sdk'
+import { Api } from '../../api'
+import {
+  WithdrawalDetailsRequestRecord,
+} from '@/js/records/requests/withdrawal-details.record'
+import EmailGetter from '@/vue/common/EmailGetter'
+
+const HORIZON_VERSION_PREFIX = 'v3'
+
+export default {
+  name: 'incoming-withdrawal-requests',
+  components: {
+    Loader,
+    Drawer,
+    NoDataMessage,
+    CollectionLoader,
+    WithdrawalRequestDetails,
+    EmailGetter,
+  },
+
+  data: _ => ({
+    requestsHistory: [],
+    isHistoryLoaded: false,
+    isLoadingFailed: false,
+    isRequestRejected: false,
+    isDetailsDrawerShown: false,
+    selectedIndex: -1,
+    firstPageLoader: () => {},
+  }),
+  computed: {
+    ...mapGetters({
+      accountId: vuexTypes.accountId,
+    }),
+    selectedRequest () {
+      return this.requestsHistory[this.selectedIndex]
+    },
+  },
+
+  created () {
+    this.initFirstPageLoader()
+  },
+
+  methods: {
+    initFirstPageLoader () {
+      this.isHistoryLoaded = false
+      this.requestsHistory = []
+      this.firstPageLoader = this.getFirstPageLoader(this.accountId)
+    },
+
+    closeDetailsDrawer () {
+      this.isDetailsDrawerShown = false
+    },
+
+    getFirstPageLoader (accountId) {
+      return function () {
+        const endpoint = `/${HORIZON_VERSION_PREFIX}/create_withdraw_requests`
+        return Api.api.getWithSignature(endpoint, {
+          filter: {
+            reviewer: accountId,
+          },
+          include: ['request_details'],
+        })
+      }
+    },
+
+    async setHistory (data) {
+      this.requestsHistory = await Promise.all(
+        data
+          .map(request => new WithdrawalDetailsRequestRecord(request))
+          .map(async request => {
+            request.asset = await this.getAssetFromBalanceId(request)
+            return request
+          })
+      )
+      this.isHistoryLoaded = true
+    },
+
+    async extendHistory (data) {
+      this.requestsHistory = this.requestsHistory.concat(
+        await Promise.all(
+          data
+            .map(request => new WithdrawalDetailsRequestRecord(request))
+            .map(async request => {
+              request.asset = await this.getAssetFromBalanceId(request)
+              return request
+            })
+        )
+      )
+    },
+
+    async getAssetFromBalanceId (request) {
+      const { _rawResponse: token } = await Api.api
+        .get(`balances/${request.balanceId}/asset`)
+      return token.data.code
+    },
+
+    showRequestDetails (index) {
+      this.selectedIndex = index
+      this.isDetailsDrawerShown = true
+    },
+
+    async approveRequest () {
+      this.isRequestRejected = true
```

Comment:
This is incorrect. Here you try to say that your component state in the `pending` state, but not in the `isRequestRejected`. So please use `isSubmitting` instead and use it [here](https://github.com/tokend/web-client/pull/143/files#diff-9d63321c02b6147139a863e3ae7972baR12). Or use another pretty `pending` name instead of `isSubmitting`, but you understand me

Reply:
Recheck your PR for similar places

Reply:
resolved